refactor(devtools): make the component metadata menu more prominent

hawkgs opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Feature


## What is the new behavior?

The menu content requires additional work but this should at least make it more prominent to the end user. 

- Add an arrow icon to indicate that this is an expansion panel. 
- Reduce the active area, which toggles the menu, to the text only. This will prevent the menu from expanding if you click the header which handles the cases when you misclick the "Show Signal Graph" button.

<table>
<tr><td>Collapsed</td><td>Expanded</td></tr>
<tr><td>
<img width="346" height="107" alt="Screenshot 2025-07-17 at 15 38 50" src="https://github.com/user-attachments/assets/8aa0d984-77d3-4bd3-b899-fd1b39af8411" />

</td><td>
<img width="336" height="149" alt="Screenshot 2025-07-18 at 10 03 05" src="https://github.com/user-attachments/assets/de4d1823-d5e3-4773-b2bf-287caf524245" />


</td></tr>
</table>

hawkgs committed:
refactor(devtools): make the component metadata menu more prominent

Add an arrow icon to indicate that this is an expansion panel. Reduce the active area (which toggles the menu) to the text only.

hawkgs commented on devtools/projects/ng-devtools/src/lib/devtools-tabs/directive-explorer/property-tab/component-metadata.component.scss:
```diff
@@ -1,5 +1,5 @@
 :host {
-  padding: 0.25rem 0.75rem;
```
Okay. I plan to revise the whole section but this is quick to do :+1:

JeanMeche replied:
I you intend to restyle it we can wait until then 👍

hawkgs replied:
Yep, that's what I meant by "additional work" in the description.

hawkgs committed:
refactor(devtools): introduce docs-ref-button

Used for external Angular docs links.

hawkgs commented:
@JeanMeche, I added the reference icons and decided to do some changes to the menu. Additionally, I've added a new commit which creates the `docs-ref-button` component that represents this same icon since there are several instances of such doc references in the app.

kirjs commented:
This PR was merged into the repository by commit cc468efc744379b0416987dcdf40d9da48b766e6.

The changes were merged into the following branches: main
